### PR TITLE
feat: docker registry updates

### PR DESCRIPTION
**Pull request by gadisn**
1. When building an image url with digest - also include the tag - this allows readability of the version being used
2. Allow defining a pull secret - in case the used docker registry requires authentication

# Pull Request Template

## Prerequisites

* If there is no issue related to your pull request - open one and assign yourself to it. If you're proposing a solution to an already opened issue - simply assign yourself to it.

## Motivation

See motivation in https://github.com/SAP/sap-btp-service-operator/issues/243

## Approach

_How does this change address the problem?_

## Pull Request status

* [x] Initial implementation
* [ ] Tested with helm template

Feel free to construct the checklist with whatever items seem most reasonable for your change. You could disassemble the Implementation part to even smaller separate checklist items if you're working on something big for example. But do make the effort to provide a checklist of some sort so that the core team as well as the community can have an idea about the progress of your Pull Request.

## Third-party code

If you use third party code with your contribution such as, components, libraries, or snippets make sure to mention that.

## Work in Progress label

For as long as development of your Pull Request is still ongoing you MUST label it with a **wip** label as well as prefix the name of the PR with *[WIP]*.

For example: *[WIP] Service brokers API*

### CODE REVIEW

File: sapbtp-operator-charts/values.yaml [hunk 1]
```diff
@@ -38,6 +38,12 @@ manager:
     req_cpu_limit: ""
   imagePullSecrets: []
 #   - name: registrySecretName
+#  imagePullSecretName: registrySecretName
```
Comment (kerenlahav) [inline sapbtp-operator-charts/values.yaml:41]: Hi Gadi, I want to test this, can you please add an example of how to set these values using --set?

Comment (gadisn) [reply sapbtp-operator-charts/values.yaml:41]: Using `--set` is a bit limited for array with inner map.
How about using instead `--values` (or `-f`) with local file which has these values?
Both `helm install` and `helm template` support it.
See for example the docs of `helm template`: https://helm.sh/docs/helm/helm_template/
Example values file: (note that the auth' line should be: "- authentication:"

manager:
  imagePullSecretName: registrySecretName
  pullSecrets: []
  - authentication:
      password: mypwd
     username: myuser
    host: myhost

Comment (kerenlahav) [reply sapbtp-operator-charts/values.yaml:41]: please review this PR
https://github.com/SAP/sap-btp-service-operator/pull/252/files

Comment (gadisn) [reply sapbtp-operator-charts/values.yaml:41]: PR 252 looks good, thanks

Comment (kerenlahav) [reply sapbtp-operator-charts/values.yaml:41]: thanks, I merged it and it will be released until EOW

**Pull request closed without merging**